department-of-veterans-affairs/abd-vro
# josiah/bie-kafka-flakey-e2e

## What was the problem?
The bie-kafka-e2e-integration-test is flakey during the CI run and can cause red herring failures due to race conditions between writing and reading data from a kafka message created and expected by the run of the integration test.

Associated tickets or Slack threads:
- #2227 

## How does this fix it?[^1]
This adds a sleep step to give the data a chance to populate before the integration test runs.

https://github.com/orgs/department-of-veterans-affairs/projects/871?pane=issue&itemId=46130417

This ticket has been created to monitor the behavior for a few sprints to verify the fix was sufficient.

## How to test this PR
- Run the github action on this branch CI: BIE Kafka End-2-End Test

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
